query-dsl/src/main/java/org/infinispan/query/dsl/impl/ValueRange.java

```diff
@@ -19,6 +19,12 @@
    private boolean includeUpper = true;
 
    public ValueRange(Object from, Object to) {
+      if (!(from instanceof Comparable)) {
+         throw new IllegalArgumentException("'from' must be an instance of java.lang.Comparable");
```

Comment:
no need to allocate a full 1000 range. You can carve out a chunk from query's allocation